### PR TITLE
`CreateSecretStorageDialog`: stop using deprecated APIs

### PR DESCRIPTION
**Pull request by richvdh**
Part of https://github.com/vector-im/crypto-internal/issues/155

First commit adds some tests. Second commit updates the component to use modern APIs.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx [hunk 1]
```diff
@@ -585,6 +585,7 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
                     <div>{_t("Enter your account password to confirm the upgrade:")}</div>
                     <div>
                         <Field
+                            id="mx_CreateSecretStorageDialog_password"
```
Comment (richvdh) [inline src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx:613]: this avoids differences in the jest snapshots depending on how many tests you run

**Comment by richvdh** [comment]
> I might prefer `[exact: false]`

TIL that was a thing

**Comment by andybalaam** [comment]
> > I might prefer `[exact: false]`
> 
> TIL that was a thing

Somehow I typoed it - it should (obviously?) be `{exact: false}` :-)

**Comment by richvdh** [comment]
> Should we add any Percy screenshots?

I've not been able to find any guidelines about when we should or should not add Percy tests, so for now I'm going with "no".

**Pull request merged**